packages/payment-processor/src/payment/index.ts

```diff
@@ -0,0 +1,101 @@
+import { ContractTransaction, Signer } from 'ethers';
+import { Provider, Web3Provider } from 'ethers/providers';
+import { bigNumberify, BigNumberish } from 'ethers/utils';
+
+import { ClientTypes, ExtensionTypes } from '@requestnetwork/types';
+
+import { getBtcPaymentUrl } from './btc-address-based';
+import { getErc20Balance, getErc20PaymentUrl, payErc20ProxyRequest } from './erc20-proxy';
+import { getEthPaymentUrl, payEthInputDataRequest } from './eth-input-data';
+import { getNetworkProvider, getProvider, getSigner } from './utils';
+
+const getPaymentNetwork = (request: ClientTypes.IRequestData): ExtensionTypes.ID | undefined => {
+  return Object.values(request.extensions).find(x => x.type === 'payment-network')?.id;
+};
+
+/**
+ * Error thrown when the network is not supported.
+ */
+export class UnsupportedNetworkError extends Error {
+  constructor(public networkName?: string) {
+    super(`Payment network ${networkName} is not supported`);
+  }
+}
+
+/**
+ * Processes a transaction to pay a Request.
+ * Supported networks: ERC20_PROXY_CONTRACT, ETH_INPUT_DATA
+ *
+ * @throws UnsupportedNetworkError if network isn't supported
+ * @param request the request to pay.
+ * @param signerOrProvider the Web3 provider, or signer. Defaults to window.ethereum.
+ */
+export async function payRequest(
+  request: ClientTypes.IRequestData,
+  signerOrProvider: Web3Provider | Signer = getProvider(),
+  amount?: BigNumberish,
+): Promise<ContractTransaction> {
+  const signer = getSigner(signerOrProvider);
+  const paymentNetwork = getPaymentNetwork(request);
+  switch (paymentNetwork) {
+    case ExtensionTypes.ID.PAYMENT_NETWORK_ERC20_PROXY_CONTRACT:
+      return payErc20ProxyRequest(request, signer, amount);
+    case ExtensionTypes.ID.PAYMENT_NETWORK_ETH_INPUT_DATA:
+      return payEthInputDataRequest(request, signer, amount);
+    default:
+      throw new UnsupportedNetworkError(paymentNetwork);
```

Comment:
Would it be nice to throw a specific error for BTC saying we can only use URL on it's case?